### PR TITLE
Upgrading to wallet connect v2

### PR DESCRIPTION
**Pull request by IanWoodard**
Since wallet connect v1 has been sunset; we must upgrade to wallet connect v2. To do this, we need to upgrade our wagmi version to 0.12.16. This upgrade requires some changes to our usage of wagmi. I did my best to keep the changes minimal and focused on keeping things similar to before while still working with the new wagmi changes.

### CODE REVIEW

File: shared/src/components/WagmiProvider.tsx [hunk 1]
```diff
@@ -8,11 +9,12 @@ import { infuraProvider } from 'wagmi/providers/infura';
...
 
 import { DEFAULT_CHAIN } from '../data/constants/Values';
+import { ALL_CHAINS } from '../data/constants/ChainSpecific';
 
 function fallbackProvider({ chainId }: { chainId?: number }) {
-  const targetChain = Object.values(chain).find((v) => v.id === chainId) || DEFAULT_CHAIN;
+  const targetChain = ALL_CHAINS.find((v) => v.id === chainId) || DEFAULT_CHAIN;
   const config = {
-    ensAddress: targetChain.ens?.address,
+    // ensAddress: targetChain.ens?.address,
```
Comment (IanWoodard) [inline shared/src/components/WagmiProvider.tsx:17]: This no longer works.

Comment (haydenshively) [reply shared/src/components/WagmiProvider.tsx:17]: :(

File: shared/src/components/WagmiProvider.tsx [hunk 2]
```diff
@@ -88,7 +89,7 @@ const client = createClient({
     }),
     new WalletConnectConnector({
       chains,
-      options: { qrcode: true },
+      options: { projectId: '3475bfb3ffc4cd0576fabe5d47efeafb' },
```
Comment (haydenshively) [inline shared/src/components/WagmiProvider.tsx:92]: Should this be a secret?

**Review by haydenshively — APPROVED** [review]

File: shared/src/components/WagmiProvider.tsx (hunk 2 above)
Comment (IanWoodard) [reply shared/src/components/WagmiProvider.tsx:92]: Not particularly, I don't think. But I moved it to env vars in the next PR.

**Pull request merged**